README.md

```diff
@@ -1,80 +1,161 @@
+$ cd sklearn_benchmarks
+$ conda env create --file dev_environment.yml
+$ conda activate sklbench_dev
+$ pip install .
+```
 
-So far, the concurrent libraries available are:
+# Quickstart
+
+The benchmark script consumes the configuration file (`config.yml` by default) which contains the specification of the scikit-learn estimators, their hyperparameters and datasets.
+
+```yml
+- name: sklearn_KNeighborsClassifier_brute_force
+  estimator: sklearn.neighbors.KNeighborsClassifier
+  strategy: hp_match
+  predict_with_onnx: True
+  parameters:
+    n_neighbors:
+      - 1
+      - 5
+      - 100
+    algorithm:
+      - brute
+    n_jobs:
+      - -1
+      - 1
+  metrics:
+    - accuracy_score
+  datasets:
+    - generator: sklearn.datasets.make_classification
+      generator_parameters:
+        n_classes: 2
+        n_redundant: 0
+      n_features: 100
+      n_samples_train:
+        - 100_000
+      n_samples_test:
+        - 1
+        - 1000
+    - generator: sklearn.datasets.make_classification
+      generator_parameters:
+        n_classes: 2
+        n_redundant: 0
+      n_features: 2
+      n_samples_train:
+        - 100_000
+      n_samples_test:
+        - 1
+        - 1000
+```
 
-- [Intel® oneAPI Data Analytics Library](https://github.com/oneapi-src/oneDAL)
-- [lightgbm](https://lightgbm.readthedocs.io/en/latest/index.html) (gradient boosting library)
-- [XGBoost](https://xgboost.readthedocs.io/en/latest/) (gradient boosting library)
-- [Catboost](https://catboost.ai/) (gradient boosting library)
-- [ONNX Runtime](https://github.com/microsoft/onnxruntime) (inferencing and training accelerator)
+**Run benchmarks for all estimators with profiling**
+```sh
+sklbench run --profile
+```
 
-Available scikit-learn's estimators are:
+**Run benchmarks for a pair of estimators with SLURM**
+```sh
+sklbench run --estimator sklearn_Ridge --estimator sklearnex_Ridge --slurm --slurm-config slurm-config.yml
+```
+Example of SLURM config file:
+```yml
+logs_dir: slurm_logs
+parameters:
+  timeout_min: 70
+  slurm_partition: normal
+  slurm_gpus_per_task: 0
+  slurm_additional_parameters:
+    hint: nomultithread
+  slurm_exclusive: True
+  slurm_cpus_per_task: 40
 
-- [KNeighborsClassifier - Brute force and KD Tree](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html)
-- [KMeans](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html)
-- [LogisticRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html)
-- [Ridge](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html)
-- [HistGradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html)
+```
+**Run HPO benchmarks only**
+```sh
+sklbench run --strategy hpo
+```
 
-## Getting Started
+**Set a custom results directory**
+```sh
+sklbench run --results-dir ~/dev/sklbench_results
+```
```

Comment:
Could you explain here which results dir is used if this is not passed explicitly? That is a timestamp will be used as result dir.

Reply:
BTW, if the user does not pass a valid timestamp, will this break the reporting tool when calling it for all the sub-folders of a given parent?

If so the reporting tool should probably be made robust to failures of timestamp parsing and just use the raw folder name as section header in the report.